[Alina Lee Hui Xiu] iP

alinaleehx opened the pull request:
Duke
-- A Personal Assistant Chatbot that helps you to keep track of various things. 

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind of having to remember things you need to do. It's,

- text-based

- easy to learn

- ~~Fast~~ _SUPER FAST_ to use

- And **FREE**!

All you need to do is,

1. download it from here.
2. add your tasks.
3. let it manage your tasks for you 😉

Features:

- [X] Managing tasks

- [X] Managing deadlines, events and todos

- [ ] And more coming soon!!

If you Java programmer, you can use it to practice Java too.
``` java
public static void main(String[] args) {
        Duke bot = new Duke("./data.txt");
        bot.start();
    }
```

Commit:
Add Gradle support

alinaleehx committed:
Implement Duke skeletal version.

Implement a skeletal version of Duke that starts by greeting the user,
simply echos commands entered by the user, and exits when the user types bye.

alinaleehx committed:
Add the ability to add and display added items.

Add the ability to store whatever text entered by the user
and display them back to the user when requested.

alinaleehx committed:
Add the ability to mark tasks as done.

alinaleehx committed:
Add support for tracking three types of tasks.

1. ToDos: tasks without any date/time attached to it
2. Deadlines: tasks that need to be done before a specific date/time
3. Events: tasks that start at a specific time and ends at a specific time

alinaleehx committed:
Use input/output redirection for testing.

Use the input/output redirection technique to semi-automate
the testing of Duke and add more comments in code.

alinaleehx committed:
Handle errors of incorrect user inputs.

alinaleehx committed:
Add support for deleting tasks from the list.

alinaleehx committed:
Use Java enums for command keywords.

alinaleehx committed:
Save the tasks in the hard disk automatically.

Save the tasks in data.txt automatically for every chat session.
Load the data from data.txt when Duke starts up.

alinaleehx committed:
Add dates for all tasks objects.

Accept dates in a yyyy-mm-dd format (e.g., 2019-10-15) and print in
a different MMM dd yyyy format (e.g., Oct 15 2019).

alinaleehx committed:
Revert "Add dates for all tasks objects."

This reverts commit 2e3c2eb38e9f9ced0034b0d75e04d6060ac7bfa4.

alinaleehx committed:
Add date to tasks objects.

Accept dates in a yyyy-mm-dd format (e.g., 2019-10-15) and print in a
different MMM dd yyyy format (e.g., Oct 15 2019).

alinaleehx committed:
Merge branch 'branch-Level-7'

alinaleehx committed:
Merge branch 'branch-Level-8'

alinaleehx committed:
Add find task command.

Give users a way to find a task by searching for a keyword.

alinaleehx committed:
Merge branch 'branch-Level-9'

alinaleehx committed:
Add JavaDoc comments to the code.

alinaleehx committed:
Make the code more OOP

Refactor the code to extract out closely related code as classes.
Ui: deals with interactions with the user.
Storage: deals with loading tasks and saving tasks in the file.
Parser: deals with making sense of the user command.
TaskList: contains the task list and has operations to add/delete tasks.

alinaleehx committed:
Merge branch 'master' into branch-A-JavaDoc.

alinaleehx committed:
Merge branch 'branch-A-JavaDoc'

wonyeji commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,138 @@
+     * @return message for all tasks matched with keyword.
+     * @throws DukeException if keyword is empty.
+     */
+    public String find(String keyword, TaskList tasks) throws DukeException{
```
I think there should be a space after `DukeException`.

wonyeji commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,138 @@
+        }
+        return tasks.find(keyword);
+    }
+}
```
I think this file is missing a newline at the end of the file? Same for the other files as well 👍

wonyeji commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Deadline class extends Task class and encapsulate a deadline task
```
```suggestion
 * The Deadline class extends the Task class and encapsulates a deadline task
```

wonyeji commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,124 @@
+import java.util.Scanner;
+
 public class Duke {
+    private Command commands = new Command();
```
Perhaps `commands` should be replaced with `command`? Since this is a single Command object.

wonyeji commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+     * @return "X" if isDone true.
+     */
+    public String getStatusIcon() {
+        return (isDone? "X" : " ");
```
```suggestion
        return isDone? "X" : " ";
```

wonyeji commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    /**
+     * Mark task as done.
```
```suggestion
     * Marks task as done.
```

wonyeji commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,138 @@
+    }
+
+    /**
+     * Return task mark as done message.
```
```suggestion
     * Return task marked as done message.
```

wonyeji commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,138 @@
+     *
+     * @param index index of task that is done.
+     * @param tasks TaskList of list of existing tasks.
+     * @return task mark as done message.
```
```suggestion
     * @return task marked as done message.
```

wonyeji commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,80 @@
+            String isDone = entry[1].trim();
+            Task task;
+            switch (taskType) {
+            case "T":
```
Rather than just the char here, perhaps you could add a char constant to the different Task classes to enhance readability and to make it easier to edit in the future when task types with overlapping char representations are created?

wonyeji commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+     * @return delete message for bot.
+     */
+    public String delete(int index) {
+        Task removed = records.remove(index);
```
Perhaps you could use a more meaningful name like `removedTask`?

wonyeji commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Prints greeting message for bot.
+     */
+    public static void greeting() {
```
I personally think `greet` would be better suited for this method name! 👍 

```suggestion
    public static void greet() {
```

wonyeji reviewed (commented):
Overall, I think your code is done well! Good job 🥇 👍  Just a few nits to fix!

alinaleehx committed:
Add a GUI to Duke.
Use the JavaFX technology to implement the GUI for Duke.

alinaleehx committed:
Merge branch 'branch-Level-10'

alinaleehx committed:
Divide classes into packages

Organize the classes into suitable java packages like duke and duke.task

alinaleehx committed:
Add JUnit tests

Add JUnit tests to test the behavior of the code. Have more than
two test methods, preferably targeting more than Command class and
Parser class.

alinaleehx committed:
Tweak the code to comply with a coding standard

Tweak the code to comply with a given coding standard.
From this point onward, any new code added are compliant with
the given coding standard.

alinaleehx committed:
Fix GUI for duke.

alinaleehx committed:
Generate a new JAR file

No jar files yet so cannot be distributed easily.

Let's package the app and the JavaFX third-party library into
an executable JAR file so that it can be distributed easily and will
work in computers that don't have JavaFX installed.

alinaleehx committed:
Add JAR file.

alinaleehx committed:
Generate a new JAR file

No jar files yet so cannot be distributed easily.

Let's package the app and the JavaFX third-party library into
an executable JAR file so that it can be distributed easily and will
work in computers that don't have JavaFX installed.

alinaleehx committed:
Use assertions

No assertions used in code to document important assumptions.

Let's use assert feature to document important assumptions that should
hold at delete, done and todo commands with appropriate messages on
assertion failure.

alinaleehx committed:
Merge pull request #1 from alinaleehx/branch-A-Assertions

Use assertions

No assertions used in code to document important assumptions. 

Let's use assert feature to document important assumptions that should 
hold at delete, done and todo commands with appropriate messages on 
assertion failure.

alinaleehx committed:
Improve searching for items

Find command returns tasks only if description matches keyword exactly.
This is not flexible and not user-friendy.

Lets add flexibility in searching for items e.g., find items even if
the keyword matches the item only partially.

alinaleehx committed:
Improve code quality

Code duplication, dead code, missing default branch in case statements
and varying level of abstraction in code fragments exists.

Let's
* Reduce code duplication by extracting common logic into another method.
* Remove dead code
* Add default branch for all case statements
* Use single level of abtraction

alinaleehx committed:
Merge pull request #2 from alinaleehx/branch-C-BetterSearch

Improve searching for items

alinaleehx committed:
Merge pull request #3 from alinaleehx/branch-A-CodeQuality

Improve code quality

alinaleehx committed:
Merge remote-tracking branch 'origin/add-gradle-support'

alinaleehx committed:
Setup Gradle to run and build Duke

Use Gradle to automate some of the build tasks of the project.

alinaleehx committed:
Merge branch 'branch-A-Gradle'

alinaleehx committed:
Improve the GUI to make it more polished

Lets:
- Make profile pictures semi-circular
- Make profile pictures more special

alinaleehx committed:
Edit return messages

alinaleehx committed:
Add a user guide

Update README and add a representative screenshot of product.

alinaleehx committed:
Create new JAR file

alinaleehx committed:
Fix README file

alinaleehx committed:
Update README

alinaleehx committed:
Fix errors in README

alinaleehx committed:
Set theme jekyll-theme-slate

alinaleehx committed:
Fix jar file

alinaleehx committed:
Merge branch 'master' of https://github.com/alinaleehx/ip

alinaleehx committed:
Improve code quality.
